dpytools/http_clients/http_custom.py

```diff
@@ -0,0 +1,69 @@
+import backoff
+import http.client
+from http.client import HTTPException
+from urllib.parse import urlparse
+import logging
+
+
+# Function to log retry attempts
+def log_retry(details):
+    logging.error(f"Request failed, retrying... Attempt #{details['tries']}")
+
+
+class HttpClient:
```

Comment:
can we rename this to `BaseHttpClient` and the file to base.py, will make it clear that this is intended to be the underlying class that api specific clients will inherit from.